server/routes/api/subscriptions.ts

```diff
@@ -0,0 +1,220 @@
+import auth from "@server/middlewares/authentication";
+import { Subscription, Event, Document } from "@server/models";
+import { authorize } from "@server/policies";
+import { presentSubscription } from "@server/presenters";
+import { SubscriptionEvent } from "@server/types";
+import { assertIn, assertPresent, assertUuid } from "@server/validation";
+import pagination from "./middlewares/pagination";
+
+const router = new Router();
+
+router.post(
+  "subscriptions.list",
+
+  auth(),
+
+  pagination(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+
+    const { documentId, event } = ctx.body;
+
+    assertPresent(documentId, "documentId is required");
+
+    // Make sure `documentId` is accompanied
+    // with valid subsribable events.
+    assertIn(
+      event,
+      ["documents.update"],
+      `${event} is not a valid subscription event for documents`
+    );
+
+    const subscriptions = await sequelize.transaction(async (transaction) => {
+      const document = await Document.findByPk(documentId, { transaction });
+
+      authorize(user, "listSubscription", document);
+
+      return Subscription.findAll({
+        where: {
+          documentId: document.id,
+          event,
+        },
+        order: [["createdAt", "DESC"]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+        transaction,
+      });
+    });
+
+    ctx.body = {
+      pagination: ctx.state.pagination,
+      data: subscriptions.map(presentSubscription),
+    };
+  }
+);
+
+router.post(
+  "subscriptions.info",
+
+  auth(),
+
+  pagination(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+
+    const subscriptions = await Subscription.findAll({
+      where: {
+        userId: user.id,
+      },
+      order: [["createdAt", "DESC"]],
+      offset: ctx.state.pagination.offset,
+      limit: ctx.state.pagination.limit,
+    });
+
+    ctx.body = {
+      pagination: ctx.state.pagination,
+      data: subscriptions.map(presentSubscription),
+    };
+  }
+);
+
+router.post(
+  "subscriptions.create",
+
+  auth(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+
+    const { documentId, event } = ctx.body;
+
+    assertPresent(documentId, "documentId is required");
+
+    // Make sure `documentId` is accompanied
+    // with valid subsribable events.
+    assertIn(
+      event,
+      ["documents.update"],
+      `${event} is not a valid subscription event for documents`
+    );
+
+    const subscription = await sequelize.transaction(async (transaction) => {
+      const document = await Document.findByPk(documentId, { transaction });
+
+      authorize(user, "createSubscription", document);
+
+      return subscriptionCreator({
+        user: user,
+        documentId: document.id,
+        event,
+        ip: ctx.request.ip,
+        transaction,
+      });
+    });
+
+    ctx.body = {
+      data: presentSubscription(subscription),
+    };
+  }
+);
+
+router.post(
```

Comment:
I'm not sure this endpoint needs to exist currently… aren't create/delete enough?